From: Jeff Mahoney <jeffm@suse.com>
Subject: hugetlb: fix section mismatches
References: fate#311831,bnc#703100
Patch-Mainline: v2.6.33-rc7

Git-commit: 094e9539bd24bbe23b8e2741e903b0f3f1f85b03

trenn: Mainline git removed __init completely. I replaced it with __meminit
(found by Mel Gorman) and will send a patch mainline.


hugetlb_sysfs_add_hstate is called by hugetlb_register_node directly
during init and also indirectly via sysfs after init.
    
This patch removes the __init tag from hugetlb_sysfs_add_hstate.
    
Signed-off-by: Jeff Mahoney <jeffm@suse.com>
Cc: Lee Schermerhorn <lee.schermerhorn@hp.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 mm/hugetlb.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6.32-SLE11-SP2/mm/hugetlb.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/mm/hugetlb.c
+++ linux-2.6.32-SLE11-SP2/mm/hugetlb.c
@@ -1454,10 +1454,10 @@ static struct attribute_group hstate_att
 	.attrs = hstate_attrs,
 };
 
-static int __init hugetlb_sysfs_add_hstate(struct hstate *h,
-				struct kobject *parent,
-				struct kobject **hstate_kobjs,
-				struct attribute_group *hstate_attr_group)
+static int hugetlb_sysfs_add_hstate(struct hstate *h,
+				    struct kobject *parent,
+				    struct kobject **hstate_kobjs,
+				    struct attribute_group *hstate_attr_group)
 {
 	int retval;
 	int hi = h - hstates;
